go/controller/internal/autogen/autogen_api_translator.go

```diff
@@ -382,37 +386,44 @@ func (a *apiTranslator) translateGroupChatForTeam(
 	}, nil
 }
 
-func (a *apiTranslator) simpleRoundRobinTeam(ctx context.Context, agent *v1alpha1.Agent, name string) (*v1alpha1.Team, error) {
+func (a *apiTranslator) simpleRoundRobinTeam(ctx context.Context, agent *v1alpha1.Agent, name, namespace string) (*v1alpha1.Team, error) {
 
 	modelConfig := a.defaultModelConfig
-	// Use the provided model config if set, otherwise use the default one
+
 	if agent.Spec.ModelConfig != "" {
-		modelConfig = types.NamespacedName{
-			Name:      agent.Spec.ModelConfig,
-			Namespace: agent.Namespace,
-		}
+		// If ModelConfig contains only the name of the ModelConfig, use the namespace of the agent.
+		modelConfig = common.GetRefFromString(agent.Spec.ModelConfig, agent.Namespace)
+	} else {
+		log.Printf("ModelConfig for Agent %s/%s not defined. Using default ModelConfig: namespace='%s', name='%s'",
+			agent.Namespace, agent.Name, modelConfig.Namespace, modelConfig.Name)
 	}
```

Comment:
This same logic about finding the modelConfig while falling back to the default is present in mutliple places, can we create a helper?